src/wp-includes/class-wp-query.php

```diff
@@ -3182,6 +3190,14 @@ public function get_posts() {
 					if ( 'ids' === $q['fields'] ) {
 						/** @var int[] */
 						$this->posts = array_map( 'intval', $cached_results['posts'] );
+
+						if ( $q['update_post_term_cache'] ) {
+							$prime_post_types = 'any' === $q['post_type'] ? $in_search_post_types : $q['post_type'];
+							update_object_term_cache( $this->posts, $prime_post_types );
+						}
+						if ( $q['update_post_meta_cache'] ) {
+							update_postmeta_cache( $this->posts );
+						}
 					} else {
 						_prime_post_caches( $cached_results['posts'], $q['update_post_term_cache'], $q['update_post_meta_cache'] );
```

Comment:
```suggestion
						_prime_post_caches( $cached_results['posts'], false, false );
```

Wonder if we could do this and move logic after else if.

Reply:
Do you mean somewhere around line 3206?